server/include/data_store.hpp

```diff
@@ -54,11 +58,16 @@ class DataStore
 
     bool RunQuery(const std::string& sql);
 
-    static int DbCallback(void* not_used, int argc, char** argv, char** azColName); // Not used
+    static int DbCallback(void* not_used, int argc, char** argv, char** azColName);
+
+    bool ReadDistinctMacAddrs(const std::string& device_id, std::vector<std::string>& mac_addrs);
```

Comment:
Can you explain what is the use of this method? From reading the method name, I would expect to read distinct MAC addresses. However this method returns a boolean and the actual MAC addresses are returned through an argument.

Why not return an std::vector<std::string> with the MAC addresses for clarity and simplicity?

Reply:
The ultimate reason for this approach is proper error reporting back to the client of the class. The return value indicates the success/failure success of the operation, while the list result is returned by ref. argument. 
I believe it is easier for the client of  the class to use the function in the manner.

Reply:
Hmm is it *really* easier though? :smile: Instead of checking a boolean we can check whether a vector is empty. Generally speaking returning a value through an argument is something very C/C++ specific which I personally dislike as deep down I am a JAVA person ( :laughing: ). I would dare say that it is more C than C++ gimmick but I have no arguments to support this right now! :smiley:

[Role expressiveness](https://blogs.msdn.microsoft.com/stevencl/2004/04/23/using-the-cognitive-dimensions-role-expressiveness/) is a common API evaluation criterion. With this in mind, I would have to insist on changing this part of the API. :stuck_out_tongue_closed_eyes:

Reply:
Fixed